posthog/clickhouse/log_entries.py

```diff
@@ -0,0 +1,68 @@
+from posthog.clickhouse.kafka_engine import KAFKA_COLUMNS, kafka_engine, ttl_period
+from posthog.clickhouse.table_engines import ReplacingMergeTree
+from posthog.kafka_client.topics import KAFKA_LOG_ENTRIES
+from posthog.settings import CLICKHOUSE_CLUSTER, CLICKHOUSE_DATABASE
+
+LOG_ENTRIES_TABLE = "log_entries"
+LOG_ENTRIES_TTL_WEEKS = 1
+
+LOG_ENTRIES_TABLE_BASE_SQL = """
+CREATE TABLE IF NOT EXISTS {table_name} ON CLUSTER '{cluster}'
+(
+    team_id UInt64,
+    log_source LowCardinality(String),
+    log_source_id VARCHAR,
+    instance_id VARCHAR,
+    timestamp DateTime64(6, 'UTC'),
+    level LowCardinality(String),
+    message VARCHAR
+    {extra_fields}
+) ENGINE = {engine}
+"""
+
+LOG_ENTRIES_TABLE_ENGINE = lambda: ReplacingMergeTree(LOG_ENTRIES_TABLE, ver="_timestamp")
+LOG_ENTRIES_TABLE_SQL = lambda: (
+    LOG_ENTRIES_TABLE_BASE_SQL
+    + """ORDER BY (team_id, log_source, log_source_id, instance_id, timestamp)
+{ttl_period}
+SETTINGS index_granularity=512
+"""
+).format(
+    table_name=LOG_ENTRIES_TABLE,
+    cluster=CLICKHOUSE_CLUSTER,
+    extra_fields=KAFKA_COLUMNS,
+    engine=LOG_ENTRIES_TABLE_ENGINE(),
+    ttl_period=ttl_period("timestamp", LOG_ENTRIES_TTL_WEEKS),
```

Comment:
I assume we'll want 90 day retention for `log_source` = 'session_replay' so it'll probably be easier to start with something like

```
TTL time + INTERVAL BATCH_LOG_TTL WEEK DELETE WHERE log_source = 'batch',
    time + INTERVAL REPLAY_LOG_TTL WEEK DELETE WHERE log_source = 'session_replay',
    time + INTERVAL DEFAULT_LOG_TTL WEEK DELETE WHERE log_source != 'session_replay' and log_source != 'batch'
```

I've totally guessed this syntax hopefully there's a simpler way of setting a default.

https://clickhouse.com/blog/using-ttl-to-manage-data-lifecycles-in-clickhouse

Reply:
Happy to start off with 90 days of TTL, batch exports doesn't really have strict requirements on that front.

Reply:
@fuziontech Maybe you have any comments on TTL from a ClickHouse perspective?

Reply:
+1 to 90 days for everything seems simpler

Reply:
90 days seems like a great place to start - If we need to we can always tune down to 30 days. See where this goes.